src/libraries/SwapMath.sol

```diff
@@ -80,9 +97,12 @@ library SwapMath {
 
             if (exactIn && sqrtRatioNextX96 != sqrtRatioTargetX96) {
                 // we didn't reach the target, so take the remainder of the maximum input as fee
-                feeAmount = uint256(-amountRemaining) - amountIn;
+                totalFeeAmount = uint256(-amountRemaining) - amountIn;
             } else {
-                feeAmount = FullMath.mulDivRoundingUp(amountIn, feePips, 1e6 - feePips);
+                if (protocolFeePips > 0) {
+                    protocolFeeAmount = FullMath.mulDivRoundingUp(amountIn, protocolFeePips, 1e6 - protocolFeePips);
+                }
+                totalFeeAmount = FullMath.mulDivRoundingUp(amountIn - protocolFeeAmount, swapFeePips, 1e6 - swapFeePips);
```

Comment:
the natspec comment indicates that `totalFeeAmount` is swap fee + protocol fee. this line doesnt smell right if `totalFeeAmount` is supposed to be the sum of both?

If it is supposed to be the sum of both fees, you're also adding protocol fees to LP fees in Pool.sol?

i.e.:
```
state.feeGrowthGlobalX128 += FullMath.mulDiv(step.totalFeeAmount, FixedPoint128.Q128, state.liquidity);
```

Reply:
It is supposed to be sum of both! First the protocol fee is taken off and then the swap fee is taken afterwards on the remaining. I'll fix it so it's not also adding the protocol fees to LP fees in Pool.sol